MediaBrowser.Common/Updates/IInstallationManager.cs

```diff
@@ -36,62 +31,63 @@ public interface IInstallationManager : IDisposable
-        /// <param name="version">The version.</param>
-        /// <returns>Task{PackageVersionInfo}.</returns>
-        Task<PackageVersionInfo> GetPackage(string name, string guid, PackageVersionClass classification, Version version);
+        /// <param name="availablePackages">The available packages.</param>
+        /// <param name="name">The name of the plugin.</param>
+        /// <param name="guid">The id of the plugin.</param>
+        /// <returns>All plugins matching the requirements.</returns>
+        IEnumerable<PackageInfo> FilterPackages(
+            IEnumerable<PackageInfo> availablePackages,
+            string name = null,
+            Guid guid = default);
 
         /// <summary>
-        /// Gets the latest compatible version.
+        /// Returns all compatible versions ordered from newest to oldest.
         /// </summary>
-        /// <param name="name">The name.</param>
-        /// <param name="guid">The assembly guid</param>
-        /// <param name="currentServerVersion">The current server version.</param>
-        /// <param name="classification">The classification.</param>
-        /// <returns>Task{PackageVersionInfo}.</returns>
-        Task<PackageVersionInfo> GetLatestCompatibleVersion(string name, string guid, Version currentServerVersion, PackageVersionClass classification = PackageVersionClass.Release);
+        /// <param name="availableVersions">The available version of the plugin.</param>
+        /// <param name="minVersion">The minimum required version of the plugin.</param>
+        /// <param name="classification">The classification of updates.</param>
+        /// <returns>All compatible versions ordered from newest to oldest.</returns>
+        IEnumerable<PackageVersionInfo> GetCompatibleVersions(
+            IEnumerable<PackageVersionInfo> availableVersions,
+            Version minVersion = null,
+            PackageVersionClass classification = PackageVersionClass.Release);
 
         /// <summary>
-        /// Gets the latest compatible version.
+        /// Returns all compatible versions ordered from newest to oldest.
         /// </summary>
         /// <param name="availablePackages">The available packages.</param>
         /// <param name="name">The name.</param>
-        /// <param name="guid">The assembly guid</param>
-        /// <param name="currentServerVersion">The current server version.</param>
+        /// <param name="guid">The guid of the plugin.</param>
+        /// <param name="minVersion">The minimum required version of the plugin.</param>
         /// <param name="classification">The classification.</param>
-        /// <returns>PackageVersionInfo.</returns>
-        PackageVersionInfo GetLatestCompatibleVersion(IEnumerable<PackageInfo> availablePackages, string name, string guid, Version currentServerVersion, PackageVersionClass classification = PackageVersionClass.Release);
+        /// <returns>All compatible versions ordered from newest to oldest.</returns>
```

Comment:
Where this sorting happens? I don't think I saw it anywhere, did I miss something?

Reply:
Json